src/fast++-read_input.cpp

```diff
@@ -1308,7 +1308,12 @@ bool read_spectra(const options_t& opts, input_state_t& state) {
     for (uint_t b : range(sflx.dims[1])) {
         fast_filter_t f;
         f.spectral = true;
-        f.id = max(state.no_filt)+1 + b;
+        if (state.no_filt.size() > 0){
```

Comment:
```suggestion
        if (!state.no_filt.empty()) {
```

Reply:
Your code was fine, this change is just for clarity. In general if we want to know whether an array is empty or not it is clearer to use ```empty()``` rather than do math with ```size()```.